gcsfs/core.py

```diff
@@ -330,16 +330,24 @@ def base(self):
     def project(self):
         return self.credentials.project
 
+    # Clean up the aiohttp session
+    #
+    # This can run from the main thread if invoked via the weakref callbcak.
+    # This can happen even if the `loop` parameter belongs to another thread
+    # (e.g. the fsspec IO worker). The control flow here is intended to attempt
+    # in-thread asynchronous cleanup first, then fallback to synchronous
+    # cleanup (which can handle cross-thread calls).
     @staticmethod
     def close_session(loop, session):
         if loop is not None and session is not None:
             if loop.is_running():
                 try:
-                    loop = asyncio.get_event_loop()
-                    loop.create_task(session.close())
+                    current_loop = asyncio.get_event_loop()
+                    current_loop.create_task(session.close())
```

Comment:
```suggestion
                    current_loop = asyncio.get_event_loop()
                    if not current_loop.is_running():
                        raise RuntimeError
                    current_loop.create_task(session.close())
```

or

```suggestion
                    current_loop = asyncio.get_running_loop()
                    current_loop.create_task(session.close())
```

Reply:
Ah, and I think this will fix the `RuntimeWarning` as well since we'll avoid creating the awaitable in the first place.

Reply:
Yup, my test shows that this address both issues. PR update incoming.